### PR TITLE
[profiler] Make profiler arg parsing more compatible with previous version.

### PR DESCRIPTION
**Pull request by kumpera**
This introduces the following changes.

By default, enable more events. Exceptions and Counters are on by default.
By default, if alloc or heapshot are enabled, we emit root events as well.

New `nodefaults` argument that disables the above changes to default arg processing.

Added "nocalls" argument for compat only purposes.

Restored "cycles/" prefix to sample frequency argument. It was already a dead argument, but kept for compat.
We restored that as existing clients pass that flag.

### CODE REVIEW

**Comment by kumpera** [comment]
CC @luhenry @rodrmoya

**Comment by luhenry** [comment]
@kumpera see https://github.com/xamarin/profiler/blob/master/src/XamarinProfiler.Core/ProfileOptions.cs#L426 for reference of what's used.

**Comment by kumpera** [comment]
Just added the last two options.

**Comment by luhenry** [comment]
@rodrmoya do we have anything that tests if we get a predefined set of profiler events when passing a predefined set of command line arguments?

File: mono/profiler/log-args.c [hunk 1]
```diff
@@ -290,6 +332,7 @@ usage (void)
 	mono_profiler_printf ("\theapshot-on-shutdown do a heapshot on runtime shutdown");
 	mono_profiler_printf ("\t                     this option is independent of the above option");
 	mono_profiler_printf ("\tcalls                enable recording enter/leave method events (very heavy)");
+	mono_profiler_printf ("\tnocalls              compat switch with previous versions of the profiler, does nothing");
```
Comment (alexrp) [inline mono/profiler/log-args.c]: I don't think we should be documenting a deprecated option.

**Comment by alexrp** [comment]
The defaults also used to include monitor events, GC handle events, and GC finalization events.

**Comment by kumpera** [comment]
We're bringing back just what the profiler needs.

**Comment by rodrmoya** [comment]
@luhenry we don't right now, but working on adding them

**Pull request merged**